apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/shareEmbedModal/personal-links-tab.tsx

```diff
@@ -0,0 +1,204 @@
+import { Button } from "@/modules/ui/components/button";
+import { DatePicker } from "@/modules/ui/components/date-picker";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/modules/ui/components/select";
+import { useTranslate } from "@tolgee/react";
+import { AlertCircleIcon, DownloadIcon } from "lucide-react";
+import Link from "next/link";
+import { useState } from "react";
+import toast from "react-hot-toast";
+import { TSegment } from "@formbricks/types/segment";
+import { generatePersonalLinksAction } from "../../actions";
+
+interface PersonalLinksTabProps {
+  environmentId: string;
+  surveyId: string;
+  segments: TSegment[];
+}
+
+// Custom DatePicker component with date restrictions
+const RestrictedDatePicker = ({
+  date,
+  updateSurveyDate,
+}: {
+  date: Date | null;
+  updateSurveyDate: (date: Date | null) => void;
+}) => {
+  // Get tomorrow's date
+  const tomorrow = new Date();
+  tomorrow.setDate(tomorrow.getDate() + 1);
+  tomorrow.setHours(0, 0, 0, 0);
+
+  const handleDateUpdate = (date: Date) => {
+    updateSurveyDate(date);
+  };
+
+  return (
+    <DatePicker
+      date={date}
+      updateSurveyDate={handleDateUpdate}
+      minDate={tomorrow}
+      onClearDate={() => updateSurveyDate(null)}
+    />
+  );
+};
+
+export const PersonalLinksTab = ({ environmentId, segments, surveyId }: PersonalLinksTabProps) => {
+  const { t } = useTranslate();
+  const [selectedSegment, setSelectedSegment] = useState<string>("");
+  const [expiryDate, setExpiryDate] = useState<Date | null>(null);
+  const [isGenerating, setIsGenerating] = useState(false);
+  const publicSegments = segments.filter((segment) => !segment.isPrivate);
+
+  // Utility function for file downloads
+  const downloadFile = (url: string, filename: string) => {
+    const link = document.createElement("a");
+    link.href = url;
+    link.download = filename;
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+  };
+
+  const handleGenerateLinks = async () => {
+    if (!selectedSegment || isGenerating) return;
+
+    setIsGenerating(true);
+
+    // Show initial toast
+    toast.loading(t("environments.surveys.summary.generating_links_toast"), {
+      duration: 5000,
+      id: "generating-links",
+    });
+
+    const result = await generatePersonalLinksAction({
+      surveyId: surveyId,
+      segmentId: selectedSegment,
+      environmentId: environmentId,
+      expirationDays: expiryDate
+        ? Math.max(1, Math.floor((expiryDate.getTime() - new Date().getTime()) / (1000 * 60 * 60 * 24)))
+        : undefined,
+    });
+
+    if (result?.data) {
+      downloadFile(result.data.downloadUrl, result.data.fileName || "personal-links.csv");
+      toast.success(t("environments.surveys.summary.links_generated_success_toast"), {
+        duration: 5000,
+        id: "generating-links",
+      });
+      setIsGenerating(false);
```

Comment:
Please move this `setIsGenerating(false);` to the end, rather than in both the if and else blocks.